internal/crypto/cert.go

```diff
@@ -59,3 +67,94 @@ func ParseCertificates(data []byte, contentType string) (certs []*x509.Certifica
 	}
 	return parsePEM(data)
 }
+
+// MergeCertificateChain returns the complete certificate chain by appending the parent certificate from the certificate bundle to the original certificate chain.
+// If the certificate bundle path is not set in the pluginConfig, it returns the original certificate chain.
+// If an error occurs while reading the certificate bundle or parsing its PEM, it returns a nil slice and a non-nil error indicating the failure reason.
+func MergeCertificateChain(pluginConfig map[string]string, originalCerts []*x509.Certificate) ([]*x509.Certificate, error) {
+	// get cert bundle path
+	if pluginConfig == nil {
+		return originalCerts, nil
+	}
+	certBundlePath, ok := pluginConfig[certBundleKey]
+	if !ok {
+		return originalCerts, nil
+	}
+
+	// validate certificate bundle path
+	basename := filepath.Base(certBundlePath)
+	if !isValidFileName(basename) {
+		return nil, fmt.Errorf("the filename of certificate bundle path is not cross-platform compatible. filename: %s", basename)
+	}
```

Comment:
It is a local config path. I'm not sure why we should consider cross platform.

Reply:
Removed.